pandas/tests/frame/test_axis_select_reindex.py

```diff
@@ -117,6 +117,26 @@ def test_drop(self):
         df.drop(labels=df[df.b > 0].index, inplace=True)
         assert_frame_equal(df, expected)
 
+    def test_drop_empty_index(self):
+        # GH 27994
+        df_index = pd.DataFrame(
+            {"a": [0, 1]},
+            index=[pd.Timestamp("2019-01-01"), pd.Timestamp("2019-01-01")],
+        )
+        assert_frame_equal(df_index.drop([]), df_index)
+        assert_frame_equal(df_index.drop(np.array([])), df_index)
+        assert_frame_equal(df_index.drop(df_index[:0]), df_index)
+
+    @pytest.mark.parametrize("labels", ["a", ["a"]])
+    def test_drop_empty_index_str(self, labels):
+        # Passing str to DataFrame.drop() for dataframe with DatetimeIndex
+        # should raise KeyError, GH 27994
+        with pytest.raises(KeyError, match="not found in axis"):
+            pd.DataFrame(
```

Comment:
create the df outside of the pytest.raises

Reply:
Changed in a new commit.